exec/java-exec/src/main/java/org/apache/drill/exec/store/easy/text/writer/TextRecordWriter.java

```diff
@@ -165,10 +182,12 @@ public void writeField() throws IOException {
   @Override
   public void cleanup() throws IOException {
     super.cleanup();
-    if (stream != null) {
-      stream.close();
-      stream = null;
-      logger.debug("closing file");
+
+    fRecordStarted = false;
+    if (writer != null) {
+      writer.close();
```

Comment:
This is a writer. The close can fail with an exception. If it does, the entire operation should fail. Should we throw a UserException` with the details here rather than let the `IOException` bubble up?`

Reply:
Good point, will update the code.